erpnext/erpnext_integrations/doctype/leaflink_settings/leaflink_settings.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import json
+import pprint
+from frappe.utils import add_days
+import datetime
+import requests
+from frappe.model.document import Document
+
+class LeafLinkSettings(Document):
+	pass
+
+
+@frappe.whitelist()
+def sync_item_with_leaflink(item_details):
+	"""
+	Sync an item in leaflink by hitting the Sync with LeafLink Button.
+	"""
+	pass
+
+
+@frappe.whitelist(allow_guest=True)
+def sync_customer_with_leaflink(**customer_data):
+	"""
+	Sync a customer with leaflink by hitting Sync with Customer Button.
+	"""
+	leaflink_settings = frappe.get_cached_doc("LeafLink Settings")
+	header = {
+		leaflink_settings.get("api_key"): leaflink_settings.get("api_value")
+	}
+
+	#check if customer exists on leaflink
+	customers_url = leaflink_settings.get("customers_url")
```

Comment:
Can we make it a POST request?

Reply:
we're fetching to see if the customer already exists on leaflink - if they do we don't make the post request